agent/unit-test/configstore_unittest.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include <configstore.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <unistd.h>
+
+class ConfigStoreTest: public ::testing::Test
+{
+public:
+    LOG_DECLARE_CLASS_CONTEXT("TEST", "Tester");
+    ConfigStoreTest() { }
+    ConfigStore *cs = new ConfigStore("");
+    std::string capNameA = "com.pelagicore.temperatureservice.gettemperature";
+    std::string capNameB = "com.pelagicore.temperatureservice.settemperature";
+    std::string capNameC = "dummyCapC";
+    std::string capNameD = "dummyCapD";
+
+    /* The Service Manifest (absolute) file paths */
+    std::string homeDir       = "/home/vagrant/softwarecontainer/agent/unit-test/";
+    std::string sm_path       = "CS_unittest_ServiceManifest.json";
+    std::string short_sm_path = "CS_unittest_short_ServiceManifest.json";
+    std::string evil_sm       = "CS_unittest_parseError.json";
+    std::string inputDir      = "/home/vagrant/";
```

Comment:
I thought about this also, but I wasn't sure how. If I had tested this from e.g. the Agent it would have felt more intuitive to test the default container paths also. More tests are needed.